src/mgr/DaemonHealthMetricCollector.cc

```diff
@@ -65,8 +65,11 @@ class PendingPGs final : public DaemonHealthMetricCollector {
   }
   bool _update(const DaemonKey& osd,
                const DaemonHealthMetric& metric) override {
-    value.n += metric.get_n();
```

Comment:
```cpp
union daemon_metric_t {
  struct {
    uint32_t n1;
    uint32_t n2;
  };
  uint64_t n;
  daemon_metric_t(uint32_t x, uint32_t y)
    : n1(x), n2(y)
  {}
  daemon_metric_t(uint64_t x = 0)
    : n(x)
  {}
};
```

Reply:
The OSD part before the changes is:

```cpp
vector<DaemonHealthMetric> OSD::get_health_metrics()
{
  vector<DaemonHealthMetric> metrics;
  // ...
  {
    std::lock_guard l(pending_creates_lock);
    auto n_primaries = pending_creates_from_mon;
    for (const auto& create : pending_creates_from_osd) {
      if (create.second) {
        n_primaries++;
      }
    }
    metrics.emplace_back(daemon_metric::PENDING_CREATING_PGS, n_primaries);
  }
```

I wonder what's going to happen during an upgrade when we could have mix of mgr and osd versions. If this is already covered, a comment would be helpful.

Reply:
@pdvian: ping.

Reply:
From code perspective, I donot see any issue with upgrade as there are no changes to daemon_metric_t as well as DaemonHealthMetric. But I am in favor of running some teuthology upgrade tests on this PR.